dataduct/s3/utils.py

```diff
@@ -142,8 +192,12 @@ def upload_dir_to_s3(s3_path, local_path, filter_function=None):
 
             key_string = os.path.join(s3_path.key, relative_path)
             key = bucket.new_key(key_string)
-            key.set_contents_from_filename(local_file_path)
+            source_size = os.stat(local_file_path).st_size
 
+            if source_size < CHUNK_SIZE:
+                key.set_contents_from_filename(local_file_path)
+            else :
+                multipart_upload(key_string, local_file_path)
```

Comment:
That is a super awesome improvement, just upload the etl stuff a few times in a loop and see how it does :+1: